Johnathan/mobs

JohnathanGlasgow opened the pull request:
This is my implementation of semi random mob generation.
Configurations of enemies and the rows they occur on are created in the editor. These configurations are represented by the class BattleTemplate.
The MobGenerator then chooses a BattleTemplate for the given row using the GenBattle method.
As discussed, the MobGenerator creates a 5 x 5 grid of potential battles (one for every potential node of the overworld tree).
However, depending on the final implementation of the data tree, it will be possible to iterate through the tree and generate mobs as needed.
I think the grid solution is fine though.

bchorne reviewed (commented):
Is there a reason that the enemies are being stored as Prefabs? Should it not be Enemy Data instead? Battle scene reads a List<Enemy.cs> from GameManager to load into already existing Entity objects in the scene.

JohnathanGlasgow commented:
This PR was opened before Enemy.cs was merged into Development with #36.

bchorne commented:
> This PR was opened before Enemy.cs was merged into Development with #36.

OOP okay yeah that'll do it.

bchorne reviewed (changes requested):
In MobManager.cs, I think having a List of ints for available rows rather than just having two ints for Minimum Row and Maximum row is overkill, the extra lists add a lot of unnecessary overhead in my opinion.

On line 61 you specifically invoke "UnityEngine.Random...", you can just leave out the UnityEngine part since we're "using UnityEngine" on all these files by default. Just a nitpick there.

You've added OverworldInfo.cs to this PR, but it looks like it's not really used for anything? Is it a holdover from when you weren't using templates?

The actual implementation itself looks fine, perhaps slightly restrictive in what enemies can go where, but those are dials we can tune later. It would also be trivial to swap over to using Enemy objects rather than prefabs after this is merged in with Dev, so that's not an issue.

JohnathanGlasgow commented:
Good call on the min/max values for rows in the Battle Template. I have implemented that.
You are right about OverworldInfo.cs being vestigial. It is now removed.
In regards to the UnityEngine.Random issue, this is to prevent ambiguity with System.Random

bchorne reviewed (approved):
I have no further issues, looks good!

Harry-Rennie commented:
This looks fine Johnathan. Inevitably there will be some tweaking on all of our ends to get our features to play nice. But what you have here is a decent standard implementation and I look forward to us fitting it all together nicely. Cheers.

And yeah @bchorne, IDE throws a tantrum when it comes to UnityEngine.Random, you will see it in my PR's as well :-)
